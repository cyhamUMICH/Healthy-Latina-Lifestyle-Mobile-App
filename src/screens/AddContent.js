import React from 'react';
import { View } from 'react-native';
import { styles } from '../styles/Styles';
import { Button } from 'react-native-elements';

const AddContent = (props) => {
  return (
    <View style={styles.app}>
      <Button 
         buttonStyle={styles.button}
         titleStyle={styles.buttonText}
         title="Add Meditation" onPress={() => props.navigation.navigate("AddMeditation", {
           topics: null,
           navigation: props.navigation
         })} 
          />
      <Button 
         buttonStyle={styles.button}
         titleStyle={styles.buttonText}
         title="Add Challenge" onPress={() => props.navigation.navigate("AddChallenge", {
           topics: null,
           navigation: props.navigation
         })} 
          />
        <Button
          buttonStyle={styles.button}
          titleStyle={styles.buttonText}
          title="Add Yoga" onPress={() => props.navigation.navigate("AddYoga", {
            topics: null,
            navigation: props.navigation
          })}
          />
           <Button
          buttonStyle={styles.button}
          titleStyle={styles.buttonText}
          title="Add Podcast" onPress={() => props.navigation.navigate("AddPodcast", {
            topics: null,
            navigation: props.navigation
          })}
          />
          <Button
          buttonStyle={styles.button}
          titleStyle={styles.buttonText}
          title="Add Course" onPress={() => props.navigation.navigate("AddCourse", {
            topics: null,
            navigation: props.navigation
          })}
          />
          <Button
          buttonStyle={styles.button}
          titleStyle={styles.buttonText}
          title="Add Journal Prompt" onPress={() => props.navigation.navigate("AddJournalPrompt", {
            topics: null,
            navigation: props.navigation
          })}
          />

    </View>
  );
};

export default AddContent;