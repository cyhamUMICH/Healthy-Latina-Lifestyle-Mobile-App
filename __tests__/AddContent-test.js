// Code from: https://github.com/byCedric/expo-guide-ci/blob/master/__tests__/App-test.js
//      and https://jestjs.io/docs/tutorial-react-native
import React from 'react';
import AddContent from '../src/screens/AddContent';
import renderer from 'react-test-renderer';

test('Renders Add Content With the Loading Screen', async () => {
  const tree = renderer.create(<AddContent />).toJSON();
  expect(tree).toMatchSnapshot();
});

test('Renders Add Content Without the Loading Screen', async () => {
  const tree = renderer.create(<AddContent skipLoadingScreen />).toJSON();
  expect(tree).toMatchSnapshot();
});