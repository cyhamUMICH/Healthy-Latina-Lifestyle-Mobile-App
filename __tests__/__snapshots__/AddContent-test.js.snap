// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Renders Add Content With the Loading Screen 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#fff",
      "flex": 1,
      "justifyContent": "center",
    }
  }
>
  <View
    style={
      Array [
        Object {
          "borderRadius": 3,
          "overflow": "hidden",
        },
        Object {
          "borderRadius": 3,
        },
        undefined,
        false,
      ]
    }
  >
    <View
      accessibilityRole="button"
      accessibilityState={
        Object {
          "busy": false,
          "disabled": false,
        }
      }
      accessible={true}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "opacity": 1,
        }
      }
    >
      <View
        style={
          Object {
            "alignItems": "center",
            "backgroundColor": "#00A89D",
            "borderColor": "#2089dc",
            "borderRadius": 3,
            "borderWidth": 0,
            "flexDirection": "row",
            "justifyContent": "center",
            "marginBottom": "2%",
            "padding": 8,
          }
        }
      >
        <Text
          style={
            Object {
              "color": "#000",
              "fontSize": 18,
              "paddingVertical": 1,
              "textAlign": "center",
            }
          }
        >
          Add Meditation
        </Text>
      </View>
    </View>
  </View>
  <View
    style={
      Array [
        Object {
          "borderRadius": 3,
          "overflow": "hidden",
        },
        Object {
          "borderRadius": 3,
        },
        undefined,
        false,
      ]
    }
  >
    <View
      accessibilityRole="button"
      accessibilityState={
        Object {
          "busy": false,
          "disabled": false,
        }
      }
      accessible={true}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "opacity": 1,
        }
      }
    >
      <View
        style={
          Object {
            "alignItems": "center",
            "backgroundColor": "#00A89D",
            "borderColor": "#2089dc",
            "borderRadius": 3,
            "borderWidth": 0,
            "flexDirection": "row",
            "justifyContent": "center",
            "marginBottom": "2%",
            "padding": 8,
          }
        }
      >
        <Text
          style={
            Object {
              "color": "#000",
              "fontSize": 18,
              "paddingVertical": 1,
              "textAlign": "center",
            }
          }
        >
          Add Challenge
        </Text>
      </View>
    </View>
  </View>
  <View
    style={
      Array [
        Object {
          "borderRadius": 3,
          "overflow": "hidden",
        },
        Object {
          "borderRadius": 3,
        },
        undefined,
        false,
      ]
    }
  >
    <View
      accessibilityRole="button"
      accessibilityState={
        Object {
          "busy": false,
          "disabled": false,
        }
      }
      accessible={true}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "opacity": 1,
        }
      }
    >
      <View
        style={
          Object {
            "alignItems": "center",
            "backgroundColor": "#00A89D",
            "borderColor": "#2089dc",
            "borderRadius": 3,
            "borderWidth": 0,
            "flexDirection": "row",
            "justifyContent": "center",
            "marginBottom": "2%",
            "padding": 8,
          }
        }
      >
        <Text
          style={
            Object {
              "color": "#000",
              "fontSize": 18,
              "paddingVertical": 1,
              "textAlign": "center",
            }
          }
        >
          Add Yoga
        </Text>
      </View>
    </View>
  </View>
  <View
    style={
      Array [
        Object {
          "borderRadius": 3,
          "overflow": "hidden",
        },
        Object {
          "borderRadius": 3,
        },
        undefined,
        false,
      ]
    }
  >
    <View
      accessibilityRole="button"
      accessibilityState={
        Object {
          "busy": false,
          "disabled": false,
        }
      }
      accessible={true}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "opacity": 1,
        }
      }
    >
      <View
        style={
          Object {
            "alignItems": "center",
            "backgroundColor": "#00A89D",
            "borderColor": "#2089dc",
            "borderRadius": 3,
            "borderWidth": 0,
            "flexDirection": "row",
            "justifyContent": "center",
            "marginBottom": "2%",
            "padding": 8,
          }
        }
      >
        <Text
          style={
            Object {
              "color": "#000",
              "fontSize": 18,
              "paddingVertical": 1,
              "textAlign": "center",
            }
          }
        >
          Add Podcast
        </Text>
      </View>
    </View>
  </View>
  <View
    style={
      Array [
        Object {
          "borderRadius": 3,
          "overflow": "hidden",
        },
        Object {
          "borderRadius": 3,
        },
        undefined,
        false,
      ]
    }
  >
    <View
      accessibilityRole="button"
      accessibilityState={
        Object {
          "busy": false,
          "disabled": false,
        }
      }
      accessible={true}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "opacity": 1,
        }
      }
    >
      <View
        style={
          Object {
            "alignItems": "center",
            "backgroundColor": "#00A89D",
            "borderColor": "#2089dc",
            "borderRadius": 3,
            "borderWidth": 0,
            "flexDirection": "row",
            "justifyContent": "center",
            "marginBottom": "2%",
            "padding": 8,
          }
        }
      >
        <Text
          style={
            Object {
              "color": "#000",
              "fontSize": 18,
              "paddingVertical": 1,
              "textAlign": "center",
            }
          }
        >
          Add Course
        </Text>
      </View>
    </View>
  </View>
  <View
    style={
      Array [
        Object {
          "borderRadius": 3,
          "overflow": "hidden",
        },
        Object {
          "borderRadius": 3,
        },
        undefined,
        false,
      ]
    }
  >
    <View
      accessibilityRole="button"
      accessibilityState={
        Object {
          "busy": false,
          "disabled": false,
        }
      }
      accessible={true}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "opacity": 1,
        }
      }
    >
      <View
        style={
          Object {
            "alignItems": "center",
            "backgroundColor": "#00A89D",
            "borderColor": "#2089dc",
            "borderRadius": 3,
            "borderWidth": 0,
            "flexDirection": "row",
            "justifyContent": "center",
            "marginBottom": "2%",
            "padding": 8,
          }
        }
      >
        <Text
          style={
            Object {
              "color": "#000",
              "fontSize": 18,
              "paddingVertical": 1,
              "textAlign": "center",
            }
          }
        >
          Add Journal Prompt
        </Text>
      </View>
    </View>
  </View>
</View>
`;

exports[`Renders Add Content Without the Loading Screen 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#fff",
      "flex": 1,
      "justifyContent": "center",
    }
  }
>
  <View
    style={
      Array [
        Object {
          "borderRadius": 3,
          "overflow": "hidden",
        },
        Object {
          "borderRadius": 3,
        },
        undefined,
        false,
      ]
    }
  >
    <View
      accessibilityRole="button"
      accessibilityState={
        Object {
          "busy": false,
          "disabled": false,
        }
      }
      accessible={true}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "opacity": 1,
        }
      }
    >
      <View
        style={
          Object {
            "alignItems": "center",
            "backgroundColor": "#00A89D",
            "borderColor": "#2089dc",
            "borderRadius": 3,
            "borderWidth": 0,
            "flexDirection": "row",
            "justifyContent": "center",
            "marginBottom": "2%",
            "padding": 8,
          }
        }
      >
        <Text
          style={
            Object {
              "color": "#000",
              "fontSize": 18,
              "paddingVertical": 1,
              "textAlign": "center",
            }
          }
        >
          Add Meditation
        </Text>
      </View>
    </View>
  </View>
  <View
    style={
      Array [
        Object {
          "borderRadius": 3,
          "overflow": "hidden",
        },
        Object {
          "borderRadius": 3,
        },
        undefined,
        false,
      ]
    }
  >
    <View
      accessibilityRole="button"
      accessibilityState={
        Object {
          "busy": false,
          "disabled": false,
        }
      }
      accessible={true}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "opacity": 1,
        }
      }
    >
      <View
        style={
          Object {
            "alignItems": "center",
            "backgroundColor": "#00A89D",
            "borderColor": "#2089dc",
            "borderRadius": 3,
            "borderWidth": 0,
            "flexDirection": "row",
            "justifyContent": "center",
            "marginBottom": "2%",
            "padding": 8,
          }
        }
      >
        <Text
          style={
            Object {
              "color": "#000",
              "fontSize": 18,
              "paddingVertical": 1,
              "textAlign": "center",
            }
          }
        >
          Add Challenge
        </Text>
      </View>
    </View>
  </View>
  <View
    style={
      Array [
        Object {
          "borderRadius": 3,
          "overflow": "hidden",
        },
        Object {
          "borderRadius": 3,
        },
        undefined,
        false,
      ]
    }
  >
    <View
      accessibilityRole="button"
      accessibilityState={
        Object {
          "busy": false,
          "disabled": false,
        }
      }
      accessible={true}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "opacity": 1,
        }
      }
    >
      <View
        style={
          Object {
            "alignItems": "center",
            "backgroundColor": "#00A89D",
            "borderColor": "#2089dc",
            "borderRadius": 3,
            "borderWidth": 0,
            "flexDirection": "row",
            "justifyContent": "center",
            "marginBottom": "2%",
            "padding": 8,
          }
        }
      >
        <Text
          style={
            Object {
              "color": "#000",
              "fontSize": 18,
              "paddingVertical": 1,
              "textAlign": "center",
            }
          }
        >
          Add Yoga
        </Text>
      </View>
    </View>
  </View>
  <View
    style={
      Array [
        Object {
          "borderRadius": 3,
          "overflow": "hidden",
        },
        Object {
          "borderRadius": 3,
        },
        undefined,
        false,
      ]
    }
  >
    <View
      accessibilityRole="button"
      accessibilityState={
        Object {
          "busy": false,
          "disabled": false,
        }
      }
      accessible={true}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "opacity": 1,
        }
      }
    >
      <View
        style={
          Object {
            "alignItems": "center",
            "backgroundColor": "#00A89D",
            "borderColor": "#2089dc",
            "borderRadius": 3,
            "borderWidth": 0,
            "flexDirection": "row",
            "justifyContent": "center",
            "marginBottom": "2%",
            "padding": 8,
          }
        }
      >
        <Text
          style={
            Object {
              "color": "#000",
              "fontSize": 18,
              "paddingVertical": 1,
              "textAlign": "center",
            }
          }
        >
          Add Podcast
        </Text>
      </View>
    </View>
  </View>
  <View
    style={
      Array [
        Object {
          "borderRadius": 3,
          "overflow": "hidden",
        },
        Object {
          "borderRadius": 3,
        },
        undefined,
        false,
      ]
    }
  >
    <View
      accessibilityRole="button"
      accessibilityState={
        Object {
          "busy": false,
          "disabled": false,
        }
      }
      accessible={true}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "opacity": 1,
        }
      }
    >
      <View
        style={
          Object {
            "alignItems": "center",
            "backgroundColor": "#00A89D",
            "borderColor": "#2089dc",
            "borderRadius": 3,
            "borderWidth": 0,
            "flexDirection": "row",
            "justifyContent": "center",
            "marginBottom": "2%",
            "padding": 8,
          }
        }
      >
        <Text
          style={
            Object {
              "color": "#000",
              "fontSize": 18,
              "paddingVertical": 1,
              "textAlign": "center",
            }
          }
        >
          Add Course
        </Text>
      </View>
    </View>
  </View>
  <View
    style={
      Array [
        Object {
          "borderRadius": 3,
          "overflow": "hidden",
        },
        Object {
          "borderRadius": 3,
        },
        undefined,
        false,
      ]
    }
  >
    <View
      accessibilityRole="button"
      accessibilityState={
        Object {
          "busy": false,
          "disabled": false,
        }
      }
      accessible={true}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "opacity": 1,
        }
      }
    >
      <View
        style={
          Object {
            "alignItems": "center",
            "backgroundColor": "#00A89D",
            "borderColor": "#2089dc",
            "borderRadius": 3,
            "borderWidth": 0,
            "flexDirection": "row",
            "justifyContent": "center",
            "marginBottom": "2%",
            "padding": 8,
          }
        }
      >
        <Text
          style={
            Object {
              "color": "#000",
              "fontSize": 18,
              "paddingVertical": 1,
              "textAlign": "center",
            }
          }
        >
          Add Journal Prompt
        </Text>
      </View>
    </View>
  </View>
</View>
`;
